### PR TITLE
4.3

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -49,6 +49,7 @@ ynh_app_setting_set --app=$app --key=domain --value=$domain
 ynh_app_setting_set --app=$app --key=path --value=$path_url
 ynh_app_setting_set --app=$app --key=admin --value=$admin
 ynh_app_setting_set --app=$app --key=password --value=$password
+ynh_app_setting_set --app=$app --key=architecture --value=$architecture
```
Comment (alexAubin) [inline scripts/install:52]: Zblerg same comment as in another app, it's weird to save this info as a setting :/

File: scripts/upgrade [hunk 2]
```diff
@@ -20,7 +20,7 @@ domain=$(ynh_app_setting_get --app=$app --key=domain)
 path_url=$(ynh_app_setting_get --app=$app --key=path)
 final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 port=$(ynh_app_setting_get --app=$app --key=port)
-architecture=$(ynh_detect_arch)
+architecture=$(ynh_app_setting_get --app=$app --key=architecture)
```
Comment (alexAubin) [inline scripts/upgrade:23]:
```suggestion
architecture=$YNH_ARCH
```

File: scripts/upgrade [hunk 3]
```diff
@@ -51,6 +51,12 @@ ynh_abort_if_errors
 #=================================================
 ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
+# If architecture doesn't exist, create it
+if [ -z "$architecture" ]; then
+	architecture=$YNH_ARCH
+	architecture=$(ynh_app_setting_get --app=$app --key=architecture)
+fi
+
```
Comment (alexAubin) [inline scripts/upgrade:59]:
```suggestion
```

**Comment by ericgaspar** [comment]
!tastemetitus

**Pull request merged**